olive/passes/onnx/insert_beam_search.py

```diff
@@ -21,14 +21,21 @@
 
 
 class InsertBeamSearch(Pass):
-    """Insert Beam Search Op."""
+    """Insert Beam Search Op. Uses WhisperBeamSearch op if ORT version >= 1.17.1, else uses BeamSearch op."""
 
     _accepts_composite_model = True
 
     @classmethod
     def _default_config(cls, accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigParam]:
         use_gpu = accelerator_spec.accelerator_type == Device.GPU
         config = {
+            "model_name": PassConfigParam(
```

Comment:
could the model_name be get from model_config or hf_config?

Reply:
yes, you can see in the description that the model_name will be inferred from the model_config is not provided. I kept this here in case they user has a model with custom tokenization and want to use it.

Reply:
for custom tokenizer case, should we provide the function to load the tokenizer in user_script? customized tokenizer might not be supported by  AutoTokenizer.from_pretrained

Reply:
From my understanding, AutoTokenizer.from_pretrained will mostly be the model supported by Hugging Face. Is that the case?

Reply:
the tokenizer with the same. It is just that they might have changed the tokenizer config json when saving the model. so as long as they provide the correct model name or path here, it will be okay.

Reply:
so we will not support https://github.com/BlinkDL/ChatRWKV/blob/main/tokenizer/rwkv_tokenizer.py?

Reply:
i was not clear, I meant if they changed the vocabulary which would in turn change the tokenization.

Reply:
the use case is similar to `model_name` here https://github.com/microsoft/Olive/blob/main/olive/passes/utils/whisper_prepost.py#L14. This is to support locally save fine-tuned hf whisper models that might have updated vocab, etc. Not to support a custom model or tokenizer class.

Reply:
for whisper, I am suppose the model_name will not be used?

Reply:
updated to remove as discussed offline.